### ISSUE TITLE
problem with type conversion?

### ISSUE THREAD

**Issue by dkoes** [gnina/libmolgrid#40]
In the below, I get an error if I don't explicitly cast to Grid4f, but this isn't an issue if types is 2D (type_vector instead of index).  Doesn't make any sense to me and don't have time to track it down right now...

```
import molgrid
import torch
import numpy as np

gmaker = molgrid.GridMaker(resolution=0.5,dimension=3.0)

coords = torch.zeros(1,3,dtype=torch.float32)
types = torch.zeros(1,dtype=torch.float32)
radii = torch.ones(1,dtype=torch.float32)
outgrid = torch.zeros(*gmaker.grid_dimensions(1),dtype=torch.float32)

gmaker.forward((0,0,0),coords,types,radii,molgrid.Grid4f(outgrid))
```

**Comment by RMeli** [comment]
I think the difference between the two signatures is that when types are 1D (`Grid<float, 1, true>`) the grid is passed by reference (`Grid<float, 4, false>& out`)
https://github.com/gnina/libmolgrid/blob/0b0f6032b54097c77b25316819273f33ac2e3e21/python/bindings.in.cpp#L806-L811

while when types are 2D (`Grid<float, 2, true>`) the grid is passed by value (`Grid<float, 4, true> g`):
https://github.com/gnina/libmolgrid/blob/0b0f6032b54097c77b25316819273f33ac2e3e21/python/bindings.in.cpp#L812-L817

**Comment by dkoes** [comment]
Appears to be fixed in v0.5.1
